File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/utils/Enviro.java
package au.com.fullcirclesolutions.saucery.utils;

public class Enviro {
    public static String SauceNativeApp() {
        return getStringVar(SauceryConstants.SAUCE_NATIVE_APP);
    }

    public static String SauceUserName() {
        return getStringVar(SauceryConstants.SAUCE_USER_NAME);
    }

    public static String SauceApiKey() {
        return getStringVar(SauceryConstants.SAUCE_API_KEY);
    }

    public static String SauceOnDemandBrowsers() {
        return getStringVar(SauceryConstants.SAUCE_ONDEMAND_BROWSERS);
    }

    public static boolean SauceUseChromeOnAndroid() {
        return getBoolVar(SauceryConstants.SAUCE_USE_CHROME_ON_ANDROID);
    }
    
    public static String BuildNumber(){
        String jenkins = JenkinsBuildNumber();
        return !IsNullOrEmpty(jenkins) ? jenkins : BambooBuildNumber();
    }

    private static String JenkinsBuildNumber() {
        return getStringVar(SauceryConstants.JENKINS_BUILD_NUMBER);
    }
    
    private static String BambooBuildNumber() {
        return getStringVar(SauceryConstants.BAMBOO_BUILD_NUMBER);
    }

    private static String getStringVar(String envVar) {
        return IsNullOrEmpty(envVar) ? null : System.getenv(envVar);
    }

    private static boolean getBoolVar(String envVar) {
        String v = getStringVar(envVar);
        return v == null ? false : Boolean.getBoolean(v);
    }

    private static int getIntVar(String envVar) {
        String v = getStringVar(envVar);
        return  v == null ? 0 : Integer.parseInt(v);
    }

    private static double getDoubleVar(String envVar) {
        String v = getStringVar(envVar);
        return v == null ? 0 : Double.parseDouble(v);
    }
    
    private static boolean IsNullOrEmpty(String str){
        return str == null || str.trim().isEmpty();
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 */

File: NUnit/Version3/UnitTests/Enviro.cs
using System;

namespace UnitTests {
    public static class Enviro {
        internal static string SauceNativeApp => GetStringVar(SauceryConstants.SAUCE_NATIVE_APP);

        internal static string SauceUserName => GetStringVar(SauceryConstants.SAUCE_USER_NAME);

        internal static string SauceApiKey => GetStringVar(SauceryConstants.SAUCE_API_KEY);

        public static string SauceOnDemandBrowsers => GetStringVar(SauceryConstants.SAUCE_ONDEMAND_BROWSERS);

        internal static bool SauceUseChromeOnAndroid => GetBoolVar(SauceryConstants.SAUCE_USE_CHROME_ON_ANDROID);

        internal static string BuildNumber {
            get {
                var jenkins = JenkinsBuildNumber;
                return !string.IsNullOrEmpty(jenkins) ? jenkins : BambooBuildNumber;
            }
        }

        public static string RecommendedAppiumVersion => GetStringVar(SauceryConstants.RECOMMENDED_APPIUM_VERSION);

        public static void SetVar(string variableName, string value) {
            if (GetStringVar(variableName) == null) {
                //Set it
                Environment.SetEnvironmentVariable(variableName, value);
            }
        }

        private static string JenkinsBuildNumber => GetStringVar(SauceryConstants.JENKINS_BUILD_NUMBER);

        private static string BambooBuildNumber => GetStringVar(SauceryConstants.BAMBOO_BUILD_NUMBER);

        private static string GetStringVar(string envVar) {
            return envVar == null ? null : Environment.GetEnvironmentVariable(envVar);
        }

        private static bool GetBoolVar(string envVar) {
            var v = GetStringVar(envVar);
            return v != null && Convert.ToBoolean(v);
        }

        private static int GetIntVar(string envVar) {
            var v = GetStringVar(envVar);
            return v == null ? 0 : Convert.ToInt32(v);
        }

        private static double GetDoubleVar(string envVar) {
            var v = GetStringVar(envVar);
            return v == null ? 0 : Convert.ToDouble(v);
        }
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 * 
 */